miio/miot_device.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+from .device import Device
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MiotDevice(Device):
+    """Main class representing a MIoT device."""
+
+    def __init__(
+        self,
+        mapping: dict,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+    ) -> None:
+        self.mapping = mapping
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+
+    def get_properties(self) -> list:
+        """Retrieve raw properties based on mapping."""
+
+        # We send property key in "did" because it's sent back via response and we can identify the property.
+        properties = [{"did": k, **v} for k, v in self.mapping.items()]
+
+        # A single request is limited to 16 properties. Therefore the
+        # properties are divided into multiple requests
+        _props = properties.copy()
```

Comment:
This note was for miio protocol, is this limitation still in place for miot? If not, this method could be greatly simplified..

Reply:
I need to check...

Reply:
Checked - still relevant